### PR TITLE
Arnej/allocate vector before use

### PR DESCRIPTION
**Pull request by arnej27959**
@geirst please review
@baldersheim @havardpe FYI

### CODE REVIEW

File: searchlib/src/vespa/searchlib/fef/termfieldmatchdata.h [hunk 1]
```diff
@@ -66,7 +67,18 @@ class TermFieldMatchData
     size_t capacity() const { return allocated() ? _data._positions._allocated : 1; }
     void reservePositions(size_t sz) {
         if (sz > capacity()) {
-            resizePositionVector(sz);
+            if (allocated()) {
+                resizePositionVector(sz);
+            } else {
+                TermFieldMatchDataPosition * n = new TermFieldMatchDataPosition[sz];
```
Comment (geirst) [inline searchlib/src/vespa/searchlib/fef/termfieldmatchdata.h]: Consider sharing allocation code with allocateVectorAndAppend() as we are basically doing the same.

**Review by geirst — COMMENTED** [review]
Overall looks good. Also consider adding explicit testing of appendPosition() to termfieldmodel_test.cpp (or other place) as I don't find any of those.

**Comment by arnej27959** [comment]
@geirst please take another look

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**